eras/conway/impl/src/Cardano/Ledger/Conway/Rules/VDel.hs

```diff
@@ -58,9 +72,44 @@ instance
   where
   type State (ConwayVDEL era) = VState era
   type Signal (ConwayVDEL era) = ConwayCommitteeCert (EraCrypto era)
-  type Environment (ConwayVDEL era) = VDelEnv era
+  type Environment (ConwayVDEL era) = PParams era
   type BaseM (ConwayVDEL era) = ShelleyBase
   type PredicateFailure (ConwayVDEL era) = ConwayVDelPredFailure era
   type Event (ConwayVDEL era) = ConwayVDelEvent era
 
-  transitionRules = undefined -- TODO
+  transitionRules = [conwayVDelTransition @era]
+
+conwayVDelTransition :: TransitionRule (ConwayVDEL era)
+conwayVDelTransition = do
+  TRC
+    ( _pp
+      , vState@VState {vsDReps, vsCCHotKeys}
+      , c
+      ) <-
+    judgmentContext
+  case c of
+    ConwayDRepReg cred -> do
+      Set.notMember cred vsDReps ?! ConwayDRepAlreadyRegisteredVDEL cred
+      -- TODO: check against a new PParam `drepDeposit`, once PParams are updated.
+      pure $
+        vState
+          { vsDReps = Set.insert cred vsDReps
+          }
+    ConwayDRepUnReg cred -> do
+      Set.member cred vsDReps ?! ConwayDRepNotRegisteredVDEL cred
+      pure $
+        vState
+          { vsDReps = Set.delete cred vsDReps
+          }
+    ConwayAuthCommitteeHotKey coldK hotK -> do
+      -- TODO: implement after the spec is to be updated w.r.t. checking if the coldK belongs to CC.
```

Comment:
The two rules here should have identical logic. They fail iff the CC member is already resigned. So `vsCCHotKeys` needs to accommodate that somebody has resigned. In the spec this is done with inserting a `nothing`, so `Map.delete` below needs to be changed.

Also, this TODO comment can be removed - we won't check is the key is in the CC. Leaving that check out actually speeds up changing the CC by an epoch.